src/tabs.ts

```diff
@@ -0,0 +1,258 @@
+                break;
+            }
+        }
+        if(!this._prev_ix)
+            this.show(0);
+    }
+    private _checkFashion():string {
+        let vw = window.innerWidth;
+        for(let style of this._fashion_style)
+            if((style.ll || 0) <= vw && (
+                    !style.ul || style.ul >= vw)) {
+                return style.val;
+            }
+        return '';
+    }
+    private _changeFashionStyle(val:string) {
+        if(this._internal_fashion != val) {
+            if(val == 'accordion')
+                this._showAccordion();
+            else
+                this._showRC();
+            this._internal_fashion = val;
+        }
+    }
+    private _resizeEvent() {
+        let val = this._checkFashion();
+        this._changeFashionStyle(val);
+        this._openActive();
+    }
+    private _showAccordion() {
+        if(this.animation) {
+            let style = document.createElement("style");
+            style.innerHTML = `${animationCSS(
+                '.animation-open', '.animation-close',
+                this._animation_styles, this.animationDur)}`
+            this.appendChild(style);
+        }
+        let title = this.querySelectorAll("[slot=tab-title]");
+        let body = this.querySelectorAll("[slot=tab-body]");
+        for (let i = 0;i < Math.min(title.length, body.length); i++) {
+            let div = document.createElement("div");
+            div.slot = "acc";
+            div.appendChild(title[i]);
+            div.appendChild(body[i]);
+            this.appendChild(div);
+        }
+    }
+    private _showRC() {
+        let acc = this.querySelectorAll("[slot=acc]");
+        for(let j of acc) {
+            this.appendChild(j.children[1]);
+            this.appendChild(j.children[0]);
+            j.remove();
+        }
+        let styles = this.querySelectorAll("style");
+        for (let s of styles)
+            s.remove();
+    }
+    private _titleClicked(e:Event) {
+        let title = this.querySelectorAll("[slot=tab-title]");
+        for(let ix = 0;ix < title.length; ix++)
+            if(title[ix] == e.target && this._prev_ix != ix)
+                this.show(ix);
+    }
+    private _animateStyles() {
+        if(!this.animation)
+            return;
+        return html`
+<style>
+    ${animationCSS(
+        '::slotted(.animation-open)',
+        '::slotted(.animation-close)',
+        this._animation_styles,
+        this.animationDur)}
+</styel>`
+    }
+    private _show(ix:number, prev_ix:number) {
+        let title = this.querySelectorAll("[slot=tab-title]");
+        let body = this.querySelectorAll("[slot=tab-body]");
+        if(prev_ix != null) {
+            this.dispatchEvent(
+                new CustomEvent("hide", { detail: { ix } })
+            );
+            if(this.animationDur) {
+                body[prev_ix]?.classList.remove("animation-open");
+                body[prev_ix]?.classList.add("animation-close");
```

Comment:
Since we are changing the animation class names for the sake of accordion, we can have that a unique private name for all the fashions and use it here.